tools/autograd/templates/Functions.cpp

```diff
@@ -25,20 +32,45 @@ Tensor maybe_multiply(const Tensor & t, const Scalar & s) {
 
 Tensor norm_backward(const Tensor & grad, const Tensor & self, const Scalar & p_) {
   auto p = p_.toDouble();
+  auto norm = self.norm(p_);
+
+  if (norm.toDouble() == 0.0) {
+    // handle case at 0 where we return a subgradient containing 0
+    return zeros_like(self);
+  }
+
   if (p == 2.0) {
-    return self * (grad / self.norm(2));
+    return self * (grad / norm);
   } else {
     auto pow_ = self.abs().pow(p - 2);
-    auto scale_v = grad / self.norm(p).toTensor().pow(p - 1);
+    auto scale_v = grad / norm.toTensor().pow(p - 1);
```

Comment:
I thought of using the C `pow` from stdlib

Reply:
Hmmm... let met try this out. Also I think we can save the scalar output (although it get's converted to a 1D Variable via `as_variable`).

Reply:
Saving the result helps, but we don't want to use `std::pow` because then we'd break double-backwards for this function.  (Tensor * Scalar would have the same problem; currently (no gradient for the Scalar).

We'll want to fix these as we get true 0-dim tensor support. (It's not an issue now because Scalars are never exposed to Python; we wrap them in 1-dim Variables).

Reply:
See #3188